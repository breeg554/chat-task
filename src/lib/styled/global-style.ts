import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html,input{
    font-family: 'Roboto', sans-serif;
    font-size: 10px;   //1rem = 10px
  }
  body,button,input{
    background-color: #fff;
    color: ${({ theme }) => theme.colors.core.primary}
  }

`;
