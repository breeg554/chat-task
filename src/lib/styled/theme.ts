export type ThemeType = typeof theme;

export const theme = {
  colors: {
    core: {
      primary: "#2C3E50",
      secondary: "#555555",
      tertiary: "rgba(0,0,0,0.25)",
      success: "#74DD88",
    },
    hover: {
      primary: "#222",
      secondary: "#444",
      tertiary: "rgba(0,0,0,0.4)",
      success: "#95ee99",
    },
  },
  gradients: {
    core: {
      primary: "linear-gradient(to right, #00c6ff, #0072ff)",
    },
    hover: {
      primary: "linear-gradient(to right, #00a5dd, #0061dd)",
    },
  },
};
