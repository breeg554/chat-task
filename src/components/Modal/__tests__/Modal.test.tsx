import { render } from "@testing-library/react";
import { Modal } from "..";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../lib/styled";
import UsersProvider from "../../../context/users";
describe("Modal component", () => {
  it("renders out and match the snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <UsersProvider>
          <Modal onClose={() => null} />
        </UsersProvider>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
