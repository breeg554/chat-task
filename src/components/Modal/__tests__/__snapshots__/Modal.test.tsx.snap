// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders out and match the snapshot 1`] = `
<div>
  <div
    class="sc-iwjdpV hhHzTJ"
  >
    <div
      class="sc-cxpSdN jOpHSD"
    >
      <div
        class="sc-llYSUQ jYuVdl"
      >
        <button
          class="sc-giYglK dTErTV"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
